### DISCUSSION TITLE
POWER control using System Commands

### DISCUSSION THREAD

**Discussion by julenba** [Arksine/moonraker#784] [Q&A]
I am trying to control the power of my printer with a SmartPlug of a not very common brand. I have managed to make a script that controls it but I am not able to run it from the mainsail interface. I would like to be able to add something like the following code on the configuration document.

```
# moonraker.conf

[power PRINTER]
type:command
TurnON: "python /home/pi/SwitchPlug/PrinterON.py"
TuroOFF: "python /home/pi/SwitchPlug/PrinterOFF.py"
```

**Comment by sk8board** [comment]
This website shows how to use python functions in your Klipper macros. Maybe this will help?

https://www.reddit.com/r/klippers/comments/za5379/extended_macro_use_python_functions_in_your_macros/

**Comment by Jaj0** [comment]
I've raised feature request for that, in the meantime you can use my solution with http server, based on other solution:
https://github.com/Jaj0/moonraker-cli-switch-server

**Comment by LittleNyanCat** [comment]
+1 for this, I'm kinda shocked that so many different protocols and smartplugs are supported, but this super simple feature that allows for all kinds of custom setups isn't
